pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -315,7 +317,7 @@ let
     optionals doBenchmark benchmarkToolDepends;
   nativeBuildInputs =
     [ ghc removeReferencesTo ] ++ optional (allPkgconfigDepends != []) (assert pkg-config != null; pkg-config) ++
-    setupHaskellDepends ++ collectedToolDepends;
+    setupHaskellDepends ++ collectedToolDepends ++ optional stdenv.hostPlatform.isGhcjs nodejs;
```

Comment:
nodejs is needed for TemplateHaskell to work, since it needs to execute compiled code at compile-time.